driver/src/test/scala/util/WithTemporaryCollection.scala

```diff
@@ -0,0 +1,18 @@
+package util
+
+import scala.util.Random
+
+import reactivemongo.api.{ Collection, CollectionMetaCommands, CollectionProducer }
+import tests.Common.timeout
+
+trait WithTemporaryCollection extends WithTemporaryDb { this: org.specs2.mutable.Specification =>
```

Comment:
What I'm afraid of, is that it tend to make some features not tested in the different kind of env/conf, whereas, even if it should be important in some case in it's (e.g. different behaviour of command according SSL/auth/version, ...).

Reply:
I'm not sure to see how that would do that? You can still run the same tests in multiple environments, right? 
As I see it, this utility just provides an isolated collection when one is needed and acceptable, either at the suite or specification level.

Reply:
Isolated collection is not an issue, but isolated DB (that for example differ in auth) can be. There is already some place where different DB is used, but we need to be careful not to make it the general rule.

Reply:
Ah, I see your point. I will parameterize it on the db, how about that?

Reply:
`Common.databases` can be used, it also include resolution of database through the slow proxy (if needed I would prefer plain function, without inheritance).

Reply:
I have parameterized the temporary collection so that each tests can choose whether to run against the common db, the slow one, or both.

I could make temporaryCollections a plain function. But for the db - it needs to be teared down after the tests, so the loaner pattern will not be enough and I think a mixin is more practical.
But then maybe it's better to have a common style for both?